src/lfortran/semantics/ast_symboltable_visitor.cpp

```diff
@@ -2397,6 +2397,9 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
                              std::queue<std::pair<std::string, std::string>>& to_be_imported_later,
                              const Location& loc) {
         ASR::symbol_t *t = m->m_symtab->resolve_symbol(remote_sym);
+        if (remote_sym == "compiler_options") {
```

Comment:
```suggestion
        if (remote_sym == "compiler_options") {
```

I'll probably check if `use iso_fortran_env, only: COMPILER_OPTION()` works with this, if it throws error you will need to convert `remote_sym` to lower case.

Reply:
The current changes works for the env ```use iso_fortran_env, only: COMPILER_OPTION```.

Reply:
I think this logic might not be correct: what if you do `use, my_module: compiler_options`, will it now skip my own function?

Reply:
Ohh I see, yes it will skip your own function. Is there no way we provide an interface in `lfortran_iso_c_binding` and then link the `compiler_options` with it somehow? If no, we need to do strict checking by using `m->m_name == "lfortran_iso_c_binding && remote_sym == "compiler_options"` something like this

Reply:
Not just that: the user can also create his own `lfortran_iso_c_binding`. We also need to ensure the `lfortran_iso_c_binding` is the intrinsic module (I think there is a flag in ASR for that in the Module).

Reply:
> Is there no way we provide an interface in `lfortran_iso_c_binding` and then link the `compiler_options` with it somehow?

We could implement this in `iso_fortran_env`, and in it call the C runtime function via `bind(c)`, but then you need to implement this in the C runtime library, not at compile time.

In fact, I think the right implementation here is to create an ASR node for this, I think the frontend is substituting the function with the string, and that should happen later. Otherwise ASR->Fortran can't easily create the original code back.